go/ql/lib/semmle/go/security/UncontrolledAllocationSize.qll

```diff
@@ -0,0 +1,34 @@
+/**
+ * Provides a taint-tracking configuration for reasoning about uncontrolled allocation size issues.
+ */
+
+import go
+
+/**
+ * Provides a taint-tracking flow for reasoning about uncontrolled allocation size issues.
+ */
+module UncontrolledAllocationSize {
+  private import UncontrolledAllocationSizeCustomizations::UncontrolledAllocationSize
+
+  /**
+   * Module for defining predicates and tracking taint flow related to uncontrolled allocation size issues.
+   */
+  module Config implements DataFlow::ConfigSig {
+    predicate isSource(DataFlow::Node source) { source instanceof Source }
+
+    predicate isSink(DataFlow::Node sink) { sink instanceof Sink }
+
+    predicate isBarrier(DataFlow::Node node) { node instanceof Sanitizer }
+
+    predicate isAdditionalFlowStep(DataFlow::Node node1, DataFlow::Node node2) {
+      exists(Function f, DataFlow::CallNode cn | cn = f.getACall() |
+        f.hasQualifiedName("strconv", ["Atoi", "ParseInt", "ParseUint", "ParseFloat"]) and
```

Comment:
I note that we have the `Strconv::IntegerParser` class, which covers 3 out of 4 of these. Is `ParseFloat` in this list because you could later cast it from a float to an integer?

I also note that there is an identical `isAdditionalFlowStep` for `DivideByZero.ql`. I don't think we should try to share the code, though, unless you can think of something very nice.

Reply:
How harmful would it be to make these models-as-data summaries (i.e. globally applicable)? I'd expect that queries not interested in numeric types already discard them in some way (even if implicitly by having incompatible sources/sinks), so the impact shouldn't be too high.

But if we did this I think it should be handled in a different PR to better analyze the impact.

Reply:
Yes, I think that would be a good idea. If you control the input to one of those functions then you control the output.